### PR TITLE
Feature/rubocop fixup

### PR DESCRIPTION
**Pull request by herebebogans**
This is mostly a rubocop autocorrect styles fixup. However I did change some easy wins on if else blocks etc.

### CODE REVIEW

File: lib/bora/resolver/credstash.rb [hunk 1]
```diff
@@ -11,31 +12,30 @@ def initialize(stack)
...
+        region
       end
 
       def parse_key_context(uri)
-        return "" if !uri.query
-        query = URI::decode_www_form(uri.query).to_h
-        context_params = query.map { |k,v| "#{k}=#{v}" }.join(" ")
+        return '' unless uri.query
+        query = URI.decode_www_form(uri.query).to_h
+        context_params = query.map { |k, v| "#{k}=#{v}" }.join(' ') # Rubocop:disabl
```
Comment (ampedandwired) [inline lib/bora/resolver/credstash.rb]: Is this supposed to be `# rubocop:disable`? Why disabled?

File: lib/bora/stack.rb [hunk 2]
```diff
@@ -180,26 +179,28 @@ def diff_parameters(cfn_options)
...
-        puts
-      end
+      return unless current_params_str || new_params_str
+      puts 'Parameters'.colorize(mode: :bold)
+      puts '----------'
+      diff = Diffy::Diff.new(current_params_str, new_params_str).to_s(String.disable_colorization ? :text : :color).chomp
+      # TODO: False positive?
+      # rubocop:disable  Lint/RequireParentheses
+      puts diff && !diff.empty? ? diff : STACK_DIFF_PARAMETERS_UNCHANGED_MESSAGE
+      # rubocop:enable Lint/RequireParentheses
```
Comment (ampedandwired) [inline lib/bora/stack.rb]: Should resolve the TODO before merging. To be honest this expression is complicated and probably warrants refactoring. Maybe something like:
```ruby
unchanged = diff.nil? || diff.empty?
puts unchanged ? STACK_DIFF_PARAMETERS_UNCHANGED_MESSAGE : diff
```

File: lib/bora/stack.rb [hunk 3]
```diff
@@ -241,73 +242,71 @@ def diff_template(context_lines, cfn_options)
...
       diff = diff.join("\n")
 
-      puts "Template".colorize(mode: :bold)
-      puts "--------"
+      puts 'Template'.colorize(mode: :bold)
+      puts '--------'
+      # TODO: False positive?
+      # rubocop:disable  Lint/RequireParentheses
       puts diff && !diff.empty? ? diff : STACK_DIFF_TEMPLATE_UNCHANGED_MESSAGE
+      # rubocop:enable  Lint/RequireParentheses
```
Comment (ampedandwired) [inline lib/bora/stack.rb]: Same comment as above

File: lib/bora/stack.rb [hunk 4]
```diff
@@ -241,73 +242,71 @@ def diff_template(context_lines, cfn_options)
...
         cfn_options[:template_body] = File.read(@template_file)
-        if !params.empty?
+        unless params.empty?
           cfn_options[:parameters] = params.map do |k, v|
             { parameter_key: k, parameter_value: v }
           end
         end
       end
-      #binding.pry
+      # binding.pry
```
Comment (ampedandwired) [inline lib/bora/stack.rb]: Remove?

File: spec/changeset_create_spec.rb [hunk 5]
```diff
@@ -1,34 +1,31 @@
...
+                                                                 creation_time: Time.parse('2016-07-21 15:01:00'),
+                                                                 changes: [
+                                                                   {
+                                                                     resource_change: {
+                                                                       action: 'Modify',
+                                                                       resource_type: 'AWS::EC2::SecurityGroup',
+                                                                       logical_resource_id: 'MySG'
+                                                                     }
+                                                                   }
+                                                                 ])
```
Comment (ampedandwired) [inline spec/changeset_create_spec.rb]: Is this really more readable?

**Comment by herebebogans** [comment]
Thanks for the feedback - agree on all points. Will update the PR.

Sorry pry keeps slipping into my PR's.

File: lib/bora/resolver/credstash.rb [hunk 6]
```diff
@@ -11,31 +12,30 @@ def initialize(stack)
...
+        raise InvalidParameter, "Invalid credstash parameter #{uri}: no credstash key" unless uri.path
         key = uri.path[1..-1]
         raise InvalidParameter, "Invalid credstash parameter #{uri}: no credstash key" if !key || key.empty?
         region = resolve_region(uri, @stack)
         context = parse_key_context(uri)
         output = `credstash --region #{region} get #{key}#{context}`
-        exit_code = $?
-        raise NotFound, output if exit_code.exitstatus != 0
+        # exit_code = $?
+        raise NotFound, output unless $CHILD_STATUS.success?
```
Comment (herebebogans) [inline lib/bora/resolver/credstash.rb]: it's required on line 3

Comment (ampedandwired) [inline lib/bora/resolver/credstash.rb]: Apologies, ctrl+f search didn't pick this up for some reason.

File: lib/bora/stack.rb (hunk 3 above)
Comment (ampedandwired) [inline lib/bora/stack.rb]: Possible to fix this before merging?

**Comment by herebebogans** [comment]
See if you like the method arg indent style now. Keeps rubocop happy and I think it's quite readble.

**Comment by ampedandwired** [comment]
Works for me, good stuff.

**Pull request merged**